rfd/0180-sso-as-mfa-method.md

```diff
@@ -0,0 +1,767 @@
+  string client_id = 2;
+  // ClientSecret is used to authenticate the client. If unset, the parent OIDC connector's
+  // ClientSecret will be used.
+  string client_secret = 3;
+  // AcrValues are Authentication Context Class Reference values. The meaning of the ACR
+  // value is context-specific and varies for identity providers. Some identity providers
+  // support MFA specific contexts, such Okta with its "phr" (phishing-resistant) ACR.
+  string acr_values = 4;
+  // Prompt is an optional OIDC prompt. An empty string omits prompt.
+  // If not specified, it defaults to select_account for backwards compatibility.
+  string prompt = 5;
+  // MaxAge is the amount of time that an IdP session is valid for. Defaults to 0 to always
+  // force re-authentication for MFA checks. This should only be set to a non-zero value if
+  // the IdP is setup to perform MFA checks on top of active user sessions.
+  google.protobuf.Duration max_age = 6;
+}
+```
+
+```diff
+message OIDCConnectorSpecV3 {
+  ...
++ // MFASettings contains settings to enabled SSO MFA checks through this auth connector.
++ OIDCConnectorMFASettings MFASettings = 19 [(gogoproto.jsontag) = "mfa_settings,omitempty"];
+}
+```
+
+**GithubConnectorMFASettings**
+
+```proto
+message GithubConnectorMFASettings {
+  // Enabled specified whether this SAML connector supports MFA checks. Defaults to false.
+  bool Enabled = 1;
+  // MFAMode specifies what MFA mode this auth connector will operate with. Defaults to "optional".
+  AuthConnectorMFAMode MFAMode = 2;
+  // ClientID is the id of the authentication client (Teleport Auth server). If unset, the parent 
+  // Github connector's ClientID will be used.
+  string client_id = 2;
+  // ClientSecret is used to authenticate the client. If unset, the parent Github connector's
+  // ClientSecret will be used.
+  string client_secret = 3;
+  // MaxAge is the amount of time that an IdP session is valid for. Defaults to 0 to always
+  // force re-authentication for MFA checks. This should only be set to a non-zero value if
+  // the IdP is setup to perform MFA checks on top of active user sessions.
+  google.protobuf.Duration max_age = 4;
+}
+```
+
+```diff
+message GithubConnectorSpecV3 {
+  ...
++ // MFASettings contains settings to enabled SSO MFA checks through this auth connector.
++ GithubConnectorMFASettings MFASettings = 10 [(gogoproto.jsontag) = "mfa_settings,omitempty"];
+}
+```
+
+### Backward Compatibility
+
+The SSO MFA flow is completely separate from the SSO login flow, meaning a new
+client cannot start an SSO login flow thinking it is doing an MFA login, or
+vice versa.
+
+For the SSO MFA flow to begin, both the client and server must pull a lever:
+1. Client (tsh, Teleport Connect, WebUI) must provide client callback URL
+2. Server be aware of the client callback URL field to return an SSO challenge
+
+As a result, there is no compatibility concerns around client/server version
+differences.
```

Comment:
The behavior will be identical to old versions: return Webauthn and/or totp challenge if the user has a webauthn or totp device.

If the server doesn't support SSO MFA, then it definitely isn't configured with an MFA enabled auth connector, so this would be the same behavior as the server being up to date without MFA enabled on the auth connector.